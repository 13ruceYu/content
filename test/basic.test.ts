import { fileURLToPath } from 'url'
import { assert, test, describe, expect } from 'vitest'
import { setup, $fetch } from '@nuxt/test-utils'

describe('fixtures:basic', async () => {
  await setup({
    rootDir: fileURLToPath(new URL('./fixtures/basic', import.meta.url)),
    server: true
  })

  test('List contents', async () => {
    const list = await $fetch('/api/_content/list')

    assert(list.length > 0)
    assert(list.includes('content:index.md'))

    // Ignored files should be listed
    assert(list.includes('content:.dot-ignored.md') === false, 'Ignored files with `.` should not be listed')
    assert(list.includes('content:-dash-ignored.md') === false, 'Ignored files with `-` should not be listed')

    assert(list.includes('fa-ir:fa:index.md') === true, 'Files with `fa-ir` prefix should be listed')
  })

  test('Get contents index', async () => {
    const index = await $fetch('/api/_content/get/content:index.md')

    expect(index).toHaveProperty('meta.mtime')
    expect(index).toHaveProperty('body')

    expect(index.body).toMatchSnapshot('basic-index-body')
  })

  test('Get ignored contents', async () => {
    const index = await $fetch('/api/_content/get/content:.ignored.md')

    expect(index).not.toHaveProperty('meta.mtime')
    expect(index).toMatchObject({})
    expect(index.body).toBeNull()
  })

  test('Get navigation', async () => {
    const list = await $fetch('/api/_content/navigation')

    expect(list).toMatchSnapshot('basic-navigation')
  })

  test('Get cats navigation', async () => {
    const list = await $fetch('/api/_content/navigation', {
      method: 'POST',
      body: {
        slug: '/cats'
      }
    })

    expect(list).toMatchSnapshot('basic-navigation-cats')
  })

  test('Get cats navigation', async () => {
    const list = await $fetch('/api/_content/navigation', {
      method: 'POST',
      body: {
        slug: '/dogs'
      }
    })

    expect(list).toMatchSnapshot('basic-navigation-dogs')
  })

  test('Search contents using `locale` helper', async () => {
    const fa = await $fetch('/locale-fa')

    expect(fa).toContain('fa-ir:fa:index.md')
    expect(fa).not.toContain('content:index.md')

    const en = await $fetch('/locale-en')

    expect(en).not.toContain('fa-ir:fa:index.md')
    expect(en).toContain('content:index.md')
  })

  test('Use default locale for unscoped contents', async () => {
    const index = await $fetch('/api/_content/get/content:index.md')

    expect(index).toHaveProperty('meta.mtime')
    expect(index).toMatchObject({
      meta: {
        locale: 'en'
      }
    })
  })
})
