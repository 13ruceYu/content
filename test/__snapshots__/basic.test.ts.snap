// Vitest Snapshot v1

exports[`fixtures:basic > Get cats navigation > basic-navigation-cats 1`] = `
[
  {
    "children": [
      {
        "draft": false,
        "id": "content:cats:index.md",
        "partial": false,
        "slug": "/cats",
        "title": "Cats list",
      },
      {
        "draft": false,
        "id": "content:cats:persian.md",
        "partial": false,
        "slug": "/cats/persian",
        "title": "Persian",
      },
      {
        "draft": false,
        "id": "content:cats:ragdoll.md",
        "partial": false,
        "slug": "/cats/ragdoll",
        "title": "Ragdoll",
      },
    ],
    "slug": "/cats",
    "title": "Cats list",
  },
]
`;

exports[`fixtures:basic > Get cats navigation > basic-navigation-dogs 1`] = `
[
  {
    "children": [
      {
        "draft": false,
        "id": "content:dogs:bulldog.md",
        "partial": false,
        "slug": "/dogs/bulldog",
        "title": "Bulldog",
      },
      {
        "draft": false,
        "id": "content:dogs:German Shepherd.md",
        "partial": false,
        "slug": "/dogs/german-shepherd",
        "title": "German Shepherd",
      },
    ],
    "slug": "/dogs",
    "title": "Dogs",
  },
]
`;

exports[`fixtures:basic > Get contents index > basic-index-body 1`] = `
{
  "children": [
    {
      "children": [
        {
          "type": "text",
          "value": "Index",
        },
      ],
      "props": {
        "id": "index",
      },
      "tag": "h1",
      "type": "element",
    },
  ],
  "toc": {
    "depth": 2,
    "links": [],
    "searchDepth": 2,
    "title": "",
  },
  "type": "root",
}
`;

exports[`fixtures:basic > Get navigation > basic-navigation 1`] = `
[
  {
    "children": [
      {
        "draft": false,
        "id": "content:cats:index.md",
        "partial": false,
        "slug": "/cats",
        "title": "Cats list",
      },
      {
        "draft": false,
        "id": "content:cats:persian.md",
        "partial": false,
        "slug": "/cats/persian",
        "title": "Persian",
      },
      {
        "draft": false,
        "id": "content:cats:ragdoll.md",
        "partial": false,
        "slug": "/cats/ragdoll",
        "title": "Ragdoll",
      },
    ],
    "id": "content:cats:index.md",
    "slug": "/cats",
    "title": "Cats list",
  },
  {
    "children": [
      {
        "draft": false,
        "id": "content:dogs:bulldog.md",
        "partial": false,
        "slug": "/dogs/bulldog",
        "title": "Bulldog",
      },
      {
        "draft": false,
        "id": "content:dogs:German Shepherd.md",
        "partial": false,
        "slug": "/dogs/german-shepherd",
        "title": "German Shepherd",
      },
    ],
    "slug": "/dogs",
    "title": "Dogs",
  },
  {
    "children": [
      {
        "draft": false,
        "id": "fa-ir:fa:index.md",
        "partial": false,
        "slug": "/",
        "title": "سلام",
      },
    ],
    "id": "fa-ir:fa:index.md",
    "slug": "/",
    "title": "سلام",
  },
  {
    "children": [
      {
        "draft": false,
        "id": "content:index.md",
        "partial": false,
        "slug": "/",
        "title": "Index",
      },
    ],
    "id": "content:index.md",
    "slug": "/",
    "title": "Index",
  },
]
`;
